node/src/components/consensus/era_supervisor/era_id.rs

```diff
@@ -0,0 +1,113 @@
+use std::{
+    fmt::{self, Debug, Display, Formatter},
+    ops::{Add, Sub},
+};
+
+use datasize::DataSize;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use casper_types::bytesrepr::{self, FromBytes, ToBytes};
+
+use crate::components::consensus::ConsensusMessage;
+
+#[derive(
+    DataSize,
+    Debug,
+    Clone,
+    Copy,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+)]
+#[serde(deny_unknown_fields)]
+pub struct EraId(pub(crate) u64);
+
+impl EraId {
+    pub(crate) fn message(self, payload: Vec<u8>) -> ConsensusMessage {
+        ConsensusMessage::Protocol {
+            era_id: self,
+            payload,
+        }
+    }
+
+    pub(crate) fn successor(self) -> EraId {
+        EraId(self.0 + 1)
+    }
+
+    /// Returns the current era minus `x`, or `None` if that would be less than `0`.
+    pub(crate) fn checked_sub(&self, x: u64) -> Option<EraId> {
```

Comment:
This could accept `EraDiff`.

Reply:
I actually find it okay to use `u64` for now, and not introduce yet another wrapper type.